apps/hash-frontend/src/pages/[shortname]/entities/[entity-uuid].page/query-editor-page.tsx

```diff
@@ -0,0 +1,193 @@
+import { GraphResolveDepths, MultiFilter } from "@blockprotocol/graph";
+import { EntityType, PropertyType } from "@blockprotocol/type-system";
+import { OntologyChip, OntologyIcon } from "@hashintel/design-system";
+import { EntityQueryEditor } from "@hashintel/query-editor";
+import { getRoots } from "@local/hash-subgraph/stdlib";
+import { Box, Typography } from "@mui/material";
+import Head from "next/head";
+import { useRouter } from "next/router";
+import { useCallback, useEffect, useState } from "react";
+
+import { useBlockProtocolQueryEntities } from "../../../../components/hooks/block-protocol-functions/knowledge/use-block-protocol-query-entities";
+import { useBlockProtocolQueryEntityTypes } from "../../../../components/hooks/block-protocol-functions/ontology/use-block-protocol-query-entity-types";
+import { useBlockProtocolQueryPropertyTypes } from "../../../../components/hooks/block-protocol-functions/ontology/use-block-protocol-query-property-types";
+import { QUERY_PROPERTY_TYPE_BASE_URL } from "./create-entity-page";
+import { EntityEditorProps } from "./entity-editor";
+import { EntityEditorContextProvider } from "./entity-editor/entity-editor-context";
+import { TypesSection } from "./entity-editor/types-section";
+import { EntityPageWrapper } from "./entity-page-wrapper";
+import { EntityPageHeader } from "./entity-page-wrapper/entity-page-header";
+
+const zeroedGraphResolveDepths: GraphResolveDepths = {
+  inheritsFrom: { outgoing: 0 },
+  constrainsValuesOn: { outgoing: 0 },
+  constrainsPropertiesOn: { outgoing: 0 },
+  constrainsLinksOn: { outgoing: 0 },
+  constrainsLinkDestinationsOn: { outgoing: 0 },
+  isOfType: { outgoing: 0 },
+  hasLeftEntity: { incoming: 0, outgoing: 0 },
+  hasRightEntity: { incoming: 0, outgoing: 0 },
+};
+
+interface QueryEditorPageProps extends EntityEditorProps {
+  entityLabel: string;
+  owner: string;
+  entityUuid: string;
+  handleSaveQuery: (value: MultiFilter) => Promise<void>;
+  mode: "create" | "edit";
+}
+
+export const QueryEditorPage = (props: QueryEditorPageProps) => {
+  const {
+    entityLabel,
+    owner,
+    entityUuid,
+    handleSaveQuery,
+    mode,
+    ...entityEditorProps
+  } = props;
+
+  const router = useRouter();
+
+  const [loading, setLoading] = useState(true);
+  const [entityTypes, setEntityTypes] = useState<EntityType[]>([]);
+  const [propertyTypes, setPropertyTypes] = useState<PropertyType[]>([]);
+  const [queryEditorKey, setQueryEditorKey] = useState(0);
+
+  const { queryEntities } = useBlockProtocolQueryEntities();
+  const { queryEntityTypes } = useBlockProtocolQueryEntityTypes();
+  const { queryPropertyTypes } = useBlockProtocolQueryPropertyTypes();
+
+  useEffect(() => {
+    const init = async () => {
+      try {
+        const { data: entityTypesSubgraph } = await queryEntityTypes({
+          data: {},
+        });
```

Comment:
Can you use `useEntityTypesOptional` instead here? It's how the data listing entity types gets loaded into the entity type editor

Reply:
Implemented here https://github.com/hashintel/hash/pull/2491/commits/b28ba382b2cf3ae85dfa76c6dd11d4023662c7c7